drivers/cas_ce_events/app/models/cas_ce_events/synthetic/event.rb

```diff
@@ -0,0 +1,65 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module CasCeEvents::Synthetic
+  class Event < ::GrdaWarehouse::Synthetic::Event
+    include ArelHelper
+
+    validates_presence_of :source
+
+    def event_date
+      source.referral_date
+    end
+
+    def event
+      case enrollment.project.ProjectType
+      when 13 # RRH
+        13
+      when 3, 10 # PSH
+        14
+      when 9 # Other PH
+        15
+      end
+    end
+
+    def referral_result
+      source.referral_result
+    end
+
+    def result_date
+      source.referral_result_date
+    end
+
+    def self.sync
+      remove_orphans
+      add_new
+    end
+
+    def self.remove_orphans
+      orphan_ids = pluck(:source_id) - CasCeEvents::GrdaWarehouse::CasReferralEvent.pluck(:id)
+      return unless orphan_ids.present?
+
+      where(source_id: orphan_ids).delete_all
```

Comment:
Are all Synthetic events always going to be tied to `CasCeEvents::GrdaWarehouse::CasReferralEvent`?  I'm probably missing something, but I wonder if the ids you are subtracting should be a method that builds up a list from potential sources.... Or, you loop over distinct source_type and find the ids of those.

Reply:
No,`GrdaWarehouse::Synthetic::Event` uses STI to differentiate between different synthetic events, but all `CasCeEvents::Synthetic::Event`s will be tied to a `CasCeEvents::GrdaWarehouse::CasReferralEvent`.